pymc_experimental/model_builder.py

```diff
@@ -60,20 +59,30 @@ def __init__(
         """
 
         super().__init__()
+        if sampler_config is None:
+            sampler_config = {}
+        if model_config is None:
+            model_config = {}
         self.model_config = model_config  # parameters for priors etc.
-        self.sample_config = sampler_config  # parameters for sampling
-        self.idata = None  # inference data object
+        self.sampler_config = sampler_config  # parameters for sampling
         self.data = data
-        self.build()
+        self.idata = (
+            None  # inference data object placeholder, idata is generated during build execution
+        )
 
-    def build(self):
+    def build(self) -> None:
         """
         Builds the defined model.
         """
 
-        with self:
-            self.build_model(self.model_config, self.data)
+        self.build_model(
+            model_instance=self,
```

Comment:
Why does this need to be passed another `self`?

Reply:
It doesn't. It's something left from very early stages of refactoring and forgot to replace it with just selfs afterwards